modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -928,6 +936,96 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
+        std::vector<int> v_inp_dims;
+        std::transform(inputWrappers.begin(), inputWrappers.end(), std::back_inserter(v_inp_dims), [] (const Ptr<VkComBackendWrapper> &w) { return (w->getMat())->dims; });
+        const int* inp_ndims = v_inp_dims.data();
+
+        // collect shapes of all input
+        std::vector<const int*> v_inp_shape;
+        std::transform(inputWrappers.begin(), inputWrappers.end(), std::back_inserter(v_inp_shape), [] (const Ptr<VkComBackendWrapper> &w) { return (w->getMat())->size.p; });
+        const int** inp_shape = v_inp_shape.data();
+
+        // collect steps of all input
+        std::vector<const size_t*> v_inp_step;
+        std::transform(inputWrappers.begin(), inputWrappers.end(), std::back_inserter(v_inp_step), [] (const Ptr<VkComBackendWrapper> &w) { return (w->getMat())->step.p; });
+        const size_t** inp_step = v_inp_step.data();
+
+        // collect info of output (ndims, shape, step)
+        // char* out = outputWrapper->getMat()->ptr<char>();
+        int out_ndims = outputWrapper->getMat()->dims;
+        const int* out_shape = outputWrapper->getMat()->size.p;
+        const size_t* out_step = outputWrapper->getMat()->step.p;
+
+        // find max ndims for broadcasting
+        int i, max_ndims = out_ndims > 2 ? out_ndims : 2;
+        for(i = 0; i < ninputs; i++)
+            max_ndims = max_ndims > inp_ndims[i] ? max_ndims : inp_ndims[i];
+
+        // buf holds the following buffers for inputs & output:
+        //  * orig_shapes, shapes (result_shape), orig_steps, steps (result_step), (ninputs+1)*4 elements in total
+        //  * ptrs, (ninputs+1)*1 elements in total
+        //  * shape_buf & step_buf, (ninputs+1)*2*max_ndims elements in total
+        //  * all_ndims, (ninputs+1)*1 elements in total
+        //  * all_type_sizes, (ninputs+1)*1 elements in total
+        AutoBuffer<size_t> buf((ninputs + 1) * (2 * max_ndims + 7));
+
+        int** orig_shapes = (int**)buf.data();
+        int** shapes = orig_shapes + ninputs + 1;
+        size_t** orig_steps = (size_t**)(shapes + ninputs + 1);
+        size_t** steps = orig_steps + ninputs + 1;
+
+        char** ptrs = (char**)(steps + ninputs + 1);
+
+        size_t* step_buf = (size_t*)(ptrs + ninputs + 1);
+        int* shape_buf = (int*)(step_buf + (ninputs + 1)*max_ndims);
+
+        int* all_ndims = shape_buf + (ninputs + 1)*max_ndims;
+        size_t* all_type_sizes = (size_t*)(all_ndims + ninputs + 1);
+
+        for(i = 0; i <= ninputs; i++) {
+            all_ndims[i] = i == 0 ? out_ndims : inp_ndims[i-1];
+            all_type_sizes[i] = i == 0 ? outputWrapper->getMat()->elemSize() : inputWrappers[i-1]->getMat()->elemSize();
+            orig_shapes[i] = (int*)(i == 0 ? out_shape : inp_shape ? inp_shape[i-1] : 0);
+            orig_steps[i] = (size_t*)(i == 0 ? out_step : inp_step ? inp_step[i-1] : 0);
+            shapes[i] = shape_buf + max_ndims*i;
+            steps[i] = step_buf + max_ndims*i;
+        }
```

Comment:
Since they share the same code with CPU implementation, could you extract and make it a function to call to reduce duplicate code?